libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -834,10 +864,13 @@ proc heartbeat(g: GossipSub) {.async.} =
 
       let peers = g.getGossipPeers()
       for peer, control in peers:
-        g.peers.withValue(peer.peerId, pubsubPeer) do:
-          g.send(
-            pubsubPeer[],
-            RPCMsg(control: some(control)))
+        # only ihave from here
+        for ihave in control.ihave:
+          if KnownLibP2PTopicsSeq.contains(ihave.topicID):
```

Comment:
also, the application should probably be able to specify the list of topics - take eth2 for example that has dynamic topics based on fork id

Reply:
you mean at runtime, by the application code?

cos atm for example my build line is:
```
make NIMFLAGS="-d:insecure -d:KnownLibP2PAgents=lighthouse,prysm,teku,nimbus,nim-libp2p -d:KnownLibP2PTopics=/eth2/3b088795/beacon_block/ssz_snappy,/eth2/b5303f2a/beacon_block/ssz_snappy" NODE_PARAMS="--rpc-address:0.0.0.0" pyrmont
```